repository/spark/docs/latest/security.md

```diff
@@ -0,0 +1,171 @@
+
+4) Check logs of the running pods:
+```bash
+$ kubectl logs spark-rpc-auth-enctryption-app-driver -f
+``` 
+5) The logs should contain auth-related messages similar to the ones in snippets below:
+```
+(driver logs):
+...
+20/02/07 16:13:09 DEBUG TransportServer: New connection accepted for remote address /10.244.0.104:46512.
+20/02/07 16:13:09 DEBUG AuthRpcHandler: Received new auth challenge for client /10.244.0.104:46512.
+20/02/07 16:13:09 DEBUG AuthRpcHandler: Authenticating challenge for app sparkSaslUser.
+20/02/07 16:13:10 DEBUG AuthEngine: Generated key with 1024 iterations in 27735 us.
+20/02/07 16:13:10 DEBUG AuthEngine: Generated key with 1024 iterations in 8148 us.
+20/02/07 16:13:10 DEBUG AuthRpcHandler: Authorization successful for client /10.244.0.104:46512.
+...
+```
+```
+(executor logs):
+...
+20/02/07 16:13:07 DEBUG TransportClientFactory: Creating new connection to spark-rpc-auth-enctryption-app-1581091973687-driver-svc.default.svc/10.244.0.103:7078
+20/02/07 16:13:07 DEBUG TransportClientFactory: Connection to spark-rpc-auth-enctryption-app-1581091973687-driver-svc.default.svc/10.244.0.103:7078 successful, running bootstraps...
+20/02/07 16:13:07 DEBUG AuthEngine: Generated key with 1024 iterations in 4715 us.
+20/02/07 16:13:08 INFO TransportClientFactory: Successfully created connection to spark-rpc-auth-enctryption-app-1581091973687-driver-svc.default.svc/10.244.0.103:7078 after 115 ms (110 ms spent in bootstraps)
+20/02/07 16:13:08 INFO CoarseGrainedExecutorBackend: Connecting to driver: spark://CoarseGrainedScheduler@spark-rpc-auth-enctryption-app-1581091973687-driver-svc.default.svc:7078
+20/02/07 16:13:08 INFO CoarseGrainedExecutorBackend: Successfully registered with driver
+...
+```
+6) Verify the network encryption:
+
+- Attach to the driver pod:
+
+```bash
+$ kubectl exec -it spark-rpc-auth-enctryption-app-driver bash
+```
+- Use `ngrep` tool to monitor the traffic on `7078` port: 
+```bash
+$ ngrep port 7078
+```
+```
+interface: eth0 (10.244.0.0/255.255.255.0)
+filter: ( port 7078 ) and ((ip || ip6) || (vlan && (ip || ip6)))
+#
+T 10.244.0.104:46480 -> 10.244.0.103:7078 [AP] #1
+  uVx....."....Qn.0....z|.4-..N.+.(E.yU..;.....5.V..e..t_L.....T....9....Zcj/A..;...7.S.....X.7.7+..Nd..x...1.Qg0D.d...vV...P V....7....\._lgi...*.#]..i..8.\...+Tu/.H.Wx*..=o.....I.K.,.....g...@:...8.;...Q...
+  .$.D..&...P.@R...I.s.M..`.oAn'.I...g.p..=....$............b.....O.|..v..:X..!H.Fot.....r83.....-Y..,X..W.......PC.....
+
+```
+
+### TLS configuration
+
+Spark allows to configure TLS for Spark web endpoints, such as Spark UI and Spark History Server UI.
+To get more information about SSL configuration in Spark , refer to the [Spark documentation](https://spark.apache.org/docs/latest/security.html#ssl-configuration).
+
+Here are the steps required to configure TLS for `SparkApplication`:
+
+1) Create a `Secret` containing all the sensitive data (passwords and key-stores):
```

Comment:
Does it make sense to add a sentence about how to create a keystone/trustore and add existing TLS keys to it?

Reply:
yes, definitely! added a note with a relevant link